legacy/reqcounter/reqcounter.go

```diff
@@ -24,28 +24,71 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// RequestCounter records the number of HTTP requests to GCR.
-type RequestCounter struct {
-	mutex    sync.Mutex // Lock to prevent race-conditions with concurrent processes.
-	requests uint64     // Number of HTTP requests since recording started.
-	since    time.Time  // When the current counter began recording requests.
+// requestCounter records the number of HTTP requests to GCR.
+// TODO: @tylerferrara in the future, add a field 'persistent bool' to determine if
+// the counter should ever reset.
+type requestCounter struct {
+	mutex    sync.Mutex    // Lock to prevent race-conditions with concurrent processes.
+	requests uint64        // Number of HTTP requests since recording started.
+	since    time.Time     // When the current counter began recording requests.
+	interval time.Duration // The duration of time between each log.
 }
 
 // log records the number of HTTP requests found and resets the counter.
-func (rc *RequestCounter) log() {
+func (rc *requestCounter) log() {
 	// Hold onto the lock when reading & writing the counter.
 	rc.mutex.Lock()
 	defer rc.mutex.Unlock()
 
 	// Log the number of requests within this measurement window.
-	msg := fmt.Sprintf("Since %s there have been %d requests to GCR.", counter.since.Format("2006-01-02 15:04:05"), rc.requests)
+	msg := fmt.Sprintf("Since %s there have been %d requests to GCR.", rc.since.Format("2006-01-02 15:04:05"), rc.requests)
 	logrus.Debug(msg)
 
 	// Reset the counter.
+	rc.reset()
+}
+
+// reset clears the request counter and stamps the current time of reset.
+// TODO: @tylerferrara in the future, use the request counter field 'persistent' to check
+// if the counter should be reset.
+func (rc *requestCounter) reset() {
 	rc.requests = 0
 	rc.since = time.Now()
 }
 
+// watch continuously logs the counter at the specified intervals.
+func (rc *requestCounter) watch() {
+	for {
+		time.Sleep(rc.interval)
+		rc.log()
+	}
+}
+
+// networkMonitor is the primary means of monitoring network traffic between CIP and GCR.
+type networkMonitor struct {
+	counters []*requestCounter
+	mutex    sync.Mutex
```

Comment:
If we have a global mutex like this, do we still need the `mutex` field in the `requestCounter` struct?

Reply:
This is a mistake on my part. That mutex should not be present. Removing.